testsuite/tests/typing-gadts/or_patterns.ml

```diff
@@ -220,8 +220,8 @@ let simple_merged_annotated_return (type a) (t : a t) (a : a) =
 Line 3, characters 12-20:
 3 |   | IntLit, (3 as x)
                 ^^^^^^^^
-Error: This pattern matches values of type a
-       This instance of a is ambiguous:
+Error: This pattern matches values of type int
+       This instance of int is ambiguous:
```

Comment:
That's a bit tricky. One way would be to add this information into the type itself (i.e. add this information to the scope field). However, there can be several equations (we shall probably only keep the ones with the highest level).
Moreover, in -principal mode this field is propagated to all children, so one may fail because of equations that are not about the node itself.
This is worth exploring in another PR.